cmd/thanos/receive.go

```diff
@@ -861,6 +865,10 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 
 	rc.tsdbMaxBlockDuration = extkingpin.ModelDuration(cmd.Flag("tsdb.max-block-duration", "Max duration for local TSDB blocks").Default("2h").Hidden())
 
+	rc.tsdbOutOfOrderTimeWindow = extkingpin.ModelDuration(cmd.Flag("tsdb.out-of-order.time-window", "[EXPERIMENTAL] Configures the allowed time window for ingestion of out-of-order samples. Disabled (0s) by default").Default("0s").Hidden())
+
+	cmd.Flag("tsdb.out-of-order.cap-max", "[EXPERIMENTAL] Configures the maximum capacity for out-of-order chunks (in samples). If set to <=0, default value 32 is assumed.").Default("0").Int64Var(&rc.tsdbOutOfOrderCapMax)
```

Comment:
This is kind of surpricing to me. 32 is the max number of samples? Seems quite a small value

Reply:
Yes, in a single chunk though. Coming from upstream here https://github.com/prometheus/prometheus/blob/main/tsdb/head.go#L160. I'm also not entirely sure how the value was chosen.

Reply:
I see. If it is single chunk then makes sense. Thanks!